### PR TITLE
Remove JRUBY_OPTS

### PR DESCRIPTION
**Pull request by chrisseaton**
See #40

### CODE REVIEW

File: ci.hocon [hunk 1]
```diff
@@ -336,7 +336,7 @@ server-benchmarks: {
 
 cext-benchmarks: ${sulong} {
   environment: {
-    JRUBY_OPTS: "-Xcexts.log.load=true",
+    TRUFFLERUBYOPTS: "-Xcexts.log.load=true",
```
Comment (eregon) [inline ci.hocon]: I think it's `TRUFFLERUBYOPT`, like `TRUFFLE` + `RUBYOPT`

File: ci.hocon [hunk 2]
```diff
@@ -380,7 +380,7 @@ test-ecosystem: ${gem-test-pack} {
 
 test-cexts: ${sulong} ${gem-test-pack} {
   environment: {
-    JRUBY_OPTS: "-J-Dgraal.TruffleCompileOnly=nothing",
+    JAVAOPTS: "-Dgraal.TruffleCompileOnly=nothing",
```
Comment (eregon) [inline ci.hocon]: JAVA_OPTS

File: tool/jt.rb [hunk 3]
```diff
@@ -772,17 +772,14 @@ def test_mri(*args)
   private :test_mri
 
   def test_compiler(*args)
-    jruby_opts = []
-
+    env = {}
+    
+    env['TRUFFLE_RUBY_OPTS'] = '-Xexceptions.print_java=true'
```
Comment (eregon) [inline tool/jt.rb]: typo

Comment (chrisseaton) [inline tool/jt.rb]: It's going to take me forever to get used to write `TRUFFLERUBYOPT`

File: tool/jt.rb [hunk 4]
```diff
@@ -884,61 +881,53 @@ def check_test_port
...
     if ENV['SL_JAR']
-      jruby_opts << '-J-cp'
-      jruby_opts << Utilities.find_sl
+      classpath << Utilities.find_sl
     end
 
-    env_vars["JRUBY_OPTS"] = jruby_opts.join(' ')
-
+    unless classpath.empty?
+      env['JAVA_OPTS'] = classpath.map { |cp| "-cp #{cp}"  }.join(' ')
```
Comment (eregon) [inline tool/jt.rb]: Actually Java accepts a single `-cp` (but the launchers accept multiple), so maybe it's simpler to just:
`"-cp #{classpath.join(':')}"`

File: tool/jt.rb [hunk 5]
```diff
@@ -884,61 +881,53 @@ def check_test_port
...
 
   def test_gems(env={}, *args)
-    env_vars   = env
-    jruby_opts = []
+    env = env.dup
 
     if ENV['GRAAL_JS_JAR']
-      jruby_opts << '-J-cp'
-      jruby_opts << Utilities.find_graal_js
+      env['JAVA_OPTS'] = "-J-cp #{Utilities.find_graal_js}"
```
Comment (eregon) [inline tool/jt.rb]: without `-J` for JAVA_OPTS

File: tool/jt.rb (hunk 3 above)
Comment (eregon) [inline tool/jt.rb]: Yeah it's quite long, but at least it's consistent with `RUBYOPT`.
Or should we only use `RUBYOPT`?

**Comment by chrisseaton** [comment]
Doesn't pass internal CI yet.

File: mx.jruby/mx_jruby.py [hunk 6]
```diff
@@ -63,41 +63,36 @@ def getResults(self):
...
-def extractArguments(cli_args):
+def extractArguments(args):
     vmArgs = []
     rubyArgs = []
     classpath = []
     print_command = False
 
-    jruby_opts = os.environ.get('JRUBY_OPTS')
-    if jruby_opts:
-        jruby_opts = jruby_opts.split(' ')
```
Comment (eregon) [inline mx.jruby/mx_jruby.py]: This probably needs to handle RUBYOPT and TRUFFLERUBYOPT, unless the Main already take care of that and Java options are disallowed in these env vars.
OTOH, I think this launcher is only used when invoking `mx ruby` directly so it should not cause failures in CI.
It simulates how GraalVM runs, but maybe we should just remove it and test with the full GraalVM when needed.

Comment (chrisseaton) [inline mx.jruby/mx_jruby.py]: `RUBYOPT` and `TRUFFLEOPT` are both handled entirely in Java code, which is a very good thing.

We should consider `JAVA_OPT` here perhaps, though.

Comment (eregon) [inline mx.jruby/mx_jruby.py]: Right, that makes sense.

**Comment by chrisseaton** [comment]
This is ready to merge. @nirvdrum can you check if your library testing code uses `JRUBY_OPTS` anywhere?

**Comment by nirvdrum** [comment]
Thanks for the heads up. I'm using `JRUBY_OPTS` to enable assertions and pass other options. I'll convert those over.

**Pull request merged**